package/MDAnalysis/core/groups.py

```diff
@@ -3324,6 +3324,66 @@ def write(self, filename=None, file_format=None,
 
         raise ValueError("No writer found for format: {}".format(filename))
 
+    def sort(self, key='ix', keyfunc=None):
+        """Return stably sorted Atomgroup by a key that specifies
+           an attribute of atomgroup. The default key is "ix".
+
+        Parameters
+        ----------
+        keys: str
+            The name for attribute of Atomgroup. (e.g. "ids", "ix")
+        keyfunc: function
+            A function that returns 1 dimension array, a key for sorting the
+            atomgroup from multiple dimension array that the attribute returns.
+
+        Returns
+        -------
+        output:
+            Sorted atomgroup.
+
+        Example
+        -------
+
+        .. code-block:: python
+
+            >>> import MDAnalysis as mda
+            >>> from MDAnalysisTests.datafiles import PDB_small
+            >>> u = mda.Universe(PDB_small)
+            >>> ag = u.atoms[[3, 2, 1, 0]]
+            >>> ag.ids
+            array([4 3 2 1])
+            >>> ag.ix
+            array([3 2 1 0])
+            >>> ag = ag.sort()
+            >>> ag.ids
+            array([1 2 3 4])
+            >>> ag.ix
+            array([0 1 2 3])
+
+        .. versionadded:: 2.0.0
+        """
+        idx = getattr(self.atoms, key)
+        if len(idx) != len(self.atoms):
+            raise ValueError("The array returned by the attribute '{}' "
+                             "must have a same length as the number of "
+                             "atoms".format(key))
+        if idx.ndim == 1:
+            order = np.argsort(idx, kind='stable')
+        elif idx.ndim > 1:
+            if keyfunc is None:
+                raise NameError("You have to assign the argument 'keyfunc' "
+                                "a proper function that returns 1D array "
+                                "as the attribute '{}' returns multiple "
+                                "dimension array.".format(key))
+            sortkeys = keyfunc(idx)
+            if sortkeys.ndim != 1:
+                raise ValueError("The function you assigned to the argument "
+                                 "'keyfunc':{} doesn't return 1D array."
+                                 .format(keyfunc))
+            order = np.argsort(sortkeys, kind='stable')
+        agsorted = self.atoms[order]
```

Comment:
nitpicking, but this seems a bit unnecessary if the next line is just to return this. It's probably understandable enough to just return `self.atoms[order]`?

Reply:
https://github.com/MDAnalysis/mdanalysis/pull/3188/commits/2f1577bfae5a9a1b166d43e7dede359059c9a7de

Fixed.